graylog2-server/src/main/java/org/graylog2/web/resources/AppConfigResource.java

```diff
@@ -18,26 +18,39 @@
 
 import com.floreysoft.jmte.Engine;
 import com.google.common.io.Resources;
+import org.apache.directory.api.util.Strings;
 import org.graylog2.Configuration;
 
 import javax.inject.Inject;
 import javax.ws.rs.GET;
 import javax.ws.rs.Path;
 import javax.ws.rs.Produces;
+import javax.ws.rs.core.Context;
+import javax.ws.rs.core.HttpHeaders;
 import javax.ws.rs.core.MediaType;
 import java.io.IOException;
+import java.net.MalformedURLException;
 import java.net.URL;
 import java.nio.charset.StandardCharsets;
 import java.util.HashMap;
+import java.util.List;
 import java.util.Map;
+import java.util.Optional;
 
 @Path("/config.js")
 public class AppConfigResource {
+    private static final String CK_OVERRIDE_SERVER_URI = "X-Graylog-Server-URL";
     private static final Engine engine = new Engine();
-    private final String content;
+    private final Configuration configuration;
 
     @Inject
     public AppConfigResource(Configuration configuration) {
+        this.configuration = configuration;
+    }
+
+    @GET
+    @Produces(MediaType.APPLICATION_OCTET_STREAM)
+    public String get(@Context HttpHeaders headers) {
         final URL templateUrl = this.getClass().getResource("/web-interface/config.js.template");
```

Comment:
Not in scope for this PR but out of interest: Why is a template being used here instead of a Jackson-annotated POJO like most other places? Should the template contain JavaScript code that goes beyond the JSON specification at one point?